sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1014,8 +1015,13 @@ def test_neighbors_metrics(n_samples=20, n_features=3,
                                                algorithm=algorithm,
                                                metric=metric, p=p,
                                                metric_params=metric_params)
-            neigh.fit(X)
-            results[algorithm] = neigh.kneighbors(test, return_distance=True)
+            feature_sl = slice(None, 2) if metric == 'haversine'\
```

Comment:
Please add a comment here explaining why we are doing this, e.g;
```py
# Haversine distance only accepts 2D data
```

also better to avoid `\` for line breaks,
```py
feature_sl = (slice(None, 2)
              if metric == 'haversine' else slice(None))
```

Reply:
Ready